shared/src/main/java/uk/gov/di/authentication/shared/helpers/PersistentIdHelper.java

```diff
@@ -27,18 +26,21 @@ public static String extractPersistentIdFromCookieHeader(Map<String, String> hea
     }
 
     public static String getExistingOrCreateNewPersistentSessionId(Map<String, String> headers) {
-        var parsedCookie = CookieHelper.parsePersistentCookie(headers);
-        if (parsedCookie.isPresent()) {
-            String existingCookieValue =
-                    InputSanitiser.sanitiseBase64(parsedCookie.get()).orElse("");
-
-            if (existingCookieValue.contains("--")) {
-                return existingCookieValue;
-            } else {
-                return appendTimestampToCookieValue(existingCookieValue);
-            }
-        } else {
-            return appendTimestampToCookieValue(IdGenerator.generate());
+        // due to change in persistent session ID format in November 2023, this could be in the old
+        // format <ID>
+        // or new format <ID>--<timestamp> e.g. U8gFw6gLpuWWHyOvj21CGphM60--1700505785611
+        var parsedOrGeneratedCookie =
+                CookieHelper.parsePersistentCookie(headers)
+                        .orElse(appendTimestampToCookieValue(IdGenerator.generate()));
+
+        if (parsedOrGeneratedCookie.contains("--")) {
```

Comment:
This assumes a raw ID without timestamp can never contain `--`. Is that true?

Reply:
No indeed. That's a faulty assumption. Does this now work for you?